Dockerfile

```diff
@@ -3,14 +3,14 @@ MAINTAINER Dimagi <devops@dimagi.com>
 
 ENV PYTHONUNBUFFERED=1 \
     PYTHONUSERBASE=/vendor \
-    PATH=/vendor/bin:$PATH
+    PATH=/vendor/bin:$PATH \
+    NODE_VERSION=5.12.0
 
-RUN curl -sL https://deb.nodesource.com/setup_5.x | bash - \
- && apt-get -y install \
-    nodejs \
- && apt-get clean \
- && rm -rf /var/lib/apt/lists/* \
- && mkdir /vendor
+RUN mkdir /vendor
+
+RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
```

Comment:
It was a compromise between reducing layers and readability. I think it makes sense to group related commands into one but I'm less keen to have completely unrelated stuff together.

I 'feel' like each RUN command should do one task (install Java, install pip requirements, install nmp etc.)